### ISSUE TITLE
On documentation formatting

### ISSUE THREAD

**Issue by kuraga** [pytorch/pytorch#190367]
(In continuation of https://discuss.pytorch.org/t/on-documentation-formatting/225171 .)

Pytorch 2.13 introduces [nn.LinearCrossEntropyLoss](https://docs.pytorch.org/docs/2.13/generated/torch.nn.LinearCrossEntropyLoss.html).
So its documentation _isn’t a legacy_ and is _thoroughly reviewed_.

There are some issues I see everywhere (e.g. _unmerged_ #146247).

1. Attributes should be `:attrs:`.
2. Classes should be `:class:`.
3. Functions should be `:func:`.
4. Python constants should be _double-backticked_.
5. Values (and mathematical intervals) should be _single-backticked_ or `:math:`.
6. Text (variable names) in math should be `\text{}`.
7. “Expressions” like `options=None`, `out_features != ()`, `vmap(grad(...))` should be ??.

What do we see in the [source](https://github.com/pytorch/pytorch/blob/v2.13.0/torch/nn/modules/loss.py#L1410)?
**Every rule** can be _not implied, without any markup_.

8. pp. 1-3 can be as p.4.
9. pp. 4 and 5 are messed (e.g. in Default: ).
10. [“Lists”](https://github.com/pytorch/pytorch/blob/v2.13.0/torch/nn/modules/loss.py#L1439-L1444) are not recognized.
11. Let’s highlight/separate Defaults?

_The issue is systematic and this is just an example._

So,

1. Do we want to follow this?
  BTW, PyTorch’s [Docstring Guidelines](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines) lists just _some_ of those rules.
  For example, [this paragraph](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines#documenting-a-class)'s example doesn't mark `Tensor` with `:class`.
2. Are there some _working_ review rules on this?
3. Do we want to lint this?
4. Is it possible to lint this?
  (https://discuss.pytorch.org/t/on-documentation-formatting/225171/4)

Any thoughts?

@pytorchbot label "module: docs"

cc @svekars @sekyondaMeta @AlannaBurke

**Comment by Skylion007** [comment]
ruff has a bunch of formatting options for documentation and autofixes for it. We should enable those @kuraga https://docs.astral.sh/ruff/rules/#pydocstyle-d and https://docs.astral.sh/ruff/rules/#pydoclint-doc
